metrics/repl.lag/lag.go

```diff
@@ -115,70 +130,122 @@ func (c *Lag) Help() blip.CollectorHelp {
+func (c *Lag) prepareBlipHeartbeatLag(levelName string, monitorID string, planName string, options map[string]string) (func(), error) {
+	if c.lagReader == nil {
+		c.dropNoHeartbeat[levelName] = !blip.Bool(options[OPT_REPORT_NO_HEARTBEAT])
 
+		table := options[OPT_HEARTBEAT_TABLE]
+		if table == "" {
+			table = blip.DEFAULT_HEARTBEAT_TABLE
+		}
 		netLatency := 50 * time.Millisecond
-		if s, ok := dom.Options[OPT_NETWORK_LATENCY]; ok {
+		if s, ok := options[OPT_NETWORK_LATENCY]; ok {
 			n, err := strconv.Atoi(s)
 			if err != nil {
-				blip.Debug("%s: invalid network-latency: %s: %s (ignoring; using default 50 ms)", plan.MonitorId, s, err)
+				blip.Debug("%s: invalid network-latency: %s: %s (ignoring; using default 50 ms)", monitorID, s, err)
 			} else {
 				netLatency = time.Duration(n) * time.Millisecond
 			}
 		}
-
-		switch dom.Options[OPT_WRITER] {
-		case "", "blip": // "" == default == blip
-			if c.lagReader == nil {
-				// Only 1 reader per plan
-				c.lagReader = heartbeat.NewBlipReader(heartbeat.BlipReaderArgs{
-					MonitorId:  plan.MonitorId,
-					DB:         c.db,
-					Table:      table,
-					SourceId:   dom.Options[OPT_SOURCE_ID],
-					SourceRole: dom.Options[OPT_SOURCE_ROLE],
-					ReplCheck:  sqlutil.CleanObjectName(dom.Options[OPT_REPL_CHECK]), // @todo sanitize better
-					Waiter: heartbeat.SlowFastWaiter{
-						MonitorId:      plan.MonitorId,
-						NetworkLatency: netLatency,
-					},
-				})
-				go c.lagReader.Start()
-				blip.Debug("%s: started reader: %s/%s (network latency: %s)", plan.MonitorId, plan.Name, level.Name, netLatency)
-			}
-			c.atLevel[level.Name] = true
-		}
+		// Only 1 reader per plan
+		c.lagReader = heartbeat.NewBlipReader(heartbeat.BlipReaderArgs{
+			MonitorId:  monitorID,
+			DB:         c.db,
+			Table:      table,
+			SourceId:   options[OPT_HEARTBEAT_SOURCE_ID],
+			SourceRole: options[OPT_HEARTBEAT_SOURCE_ROLE],
+			ReplCheck:  c.replCheck,
+			Waiter: heartbeat.SlowFastWaiter{
+				MonitorId:      monitorID,
+				NetworkLatency: netLatency,
+			},
+		})
+		go c.lagReader.Start()
+		blip.Debug("%s: started reader: %s/%s (network latency: %s)", monitorID, planName, levelName, netLatency)
 	}
-
+	c.lagSourceIn[levelName] = LAG_SOURCE_BLIP
 	var cleanup func()
 	if c.lagReader != nil {
 		cleanup = func() {
-			blip.Debug("%s: stopping reader", plan.MonitorId)
+			blip.Debug("%s: stopping reader", monitorID)
 			c.lagReader.Stop()
 		}
 	}
-
 	return cleanup, nil
 }
 
-func (c *Lag) Collect(ctx context.Context, levelName string) ([]blip.MetricValue, error) {
-	if !c.atLevel[levelName] {
-		return nil, nil
-	}
+func (c *Lag) preparePFS(levelName string) error {
+	c.lagSourceIn[levelName] = LAG_SOURCE_PFS
+
+	// Try collecting, discard metrics
+	_, err := c.collectLagFromPFS(context.TODO(), levelName)
+	return err
+}
+
+func (c *Lag) collectBlipHeartBeatLag(ctx context.Context, levelName string) ([]blip.MetricValue, error) {
```

Comment:
We can shorten all these func names and make them consistent (e.g. this one doesn't have "From" but the other one does), presuming they're not inlined:
* prepareBlip
* collectBlip
* preparePFS
* collectPFS